### PR TITLE
chore(meta): revamp of the issue guidelines content

### PR DESCRIPTION
**Pull request by dipikabh**
### Description

This pull request revamps the content of the [Issue guidelines](https://developer.mozilla.org/en-US/docs/MDN/Community/Issues) page. It merges the content of the following two pages:

- [Issue guidelines](https://developer.mozilla.org/en-US/docs/MDN/Community/Issues)
- [Issue triage on MDN Web Docs](https://developer.mozilla.org/en-US/docs/MDN/Community/Issues/Issue_triage)

### Related issues and pull requests

The base work and review for this was done here: https://github.com/mdn/temp-processes/pull/7

/cc @schalkneethling, @bsmth, @Rumyra 

### TO DO
- Move the section "Guidelines for triaging issues" after the section "Guidelines for reporting an issue" - not doing this, as per the suggestion by @bsmth to look at the content from the perspective of contributors. Keeping this in mind, updated the title and added a hint in https://github.com/mdn/content/pull/23479/commits/d13c9927356d708953bec81da9e159b7c2f81051.
- [x]  Remove the file "/en-US/mdn/community/issues/issue_triage/index.md" and redirect it to "/en-US/mdn/community/issues/index.md"

  - I am getting an error while doing this:
  `yarn content delete "MDN/Community/Issues/Issue_triage" --recursive --redirect MDN/Community/Issues`
   "error: document does not exists: /en-US/docs/MDN/Community/Issues/Issue_triage"
   "error Command failed with exit code 1."
   Getting the same error without the `--redirect` or the `--recursive` flags
  - Update: This has been fixed by @Josh-Cena in https://github.com/mdn/content/pull/23479/commits/80d4d710757eaff9f2a1a4b6cd45c3b49fc7b691

### CODE REVIEW

File: files/en-us/mdn/community/issues/index.md [hunk 1]
```diff
@@ -10,100 +10,203 @@ tags:
...
+These are the guidelines to follow while triaging each issue.
+
+#### Review if the issue is valid
+
+These are some of the things to keep in mind while reviewing the validity of an issue:
+
+- Check if the issue raised is valid and if the fix will improve the content for the readers and the website.
+- Evaluate if the impact of the fix will be small or site-wide.
+- Evaluate if the fix for the issue will need a discussion first, in which case, point the author to open a [discussion](https://github.com/mdn/mdn-community/discussions) instead.
+- Check if the issue is inline with our [writing guidelines](/en-US/docs/MDN/Writing_guidelines/Writing_style_guide) and [templates](/en-US/docs/MDN/Writing_guidelines/Page_structures/Page_types).
```
Comment (Josh-Cena) [inline files/en-us/mdn/community/issues/index.md:160]:
```suggestion
- Check if the issue is in line with our [writing guidelines](/en-US/docs/MDN/Writing_guidelines/Writing_style_guide) and [templates](/en-US/docs/MDN/Writing_guidelines/Page_structures/Page_types).
```

File: files/en-us/mdn/community/issues/index.md [hunk 2]
```diff
@@ -10,100 +10,203 @@ tags:
...
 
-```markdown
-## Remove \{{ warning }} macro from documents
+While reporting an issue or participating in a conversation in an issue, always ensure that your inputs are contributing to the overall progress of the project. Consider whether the issues you open and your comments in an issue are constructive and on topic and are not just adding noise.
 
-We should no longer be using the `\{{ warning }}` macro in our documentation.
+Do the following:
 
-### Task description
+- Before filing an issue, consider if you need to start a [discussion](https://github.com/mdn/mdn-community/discussions) in the MDN Web Docs project on GitHub. Use discussions to gain different viewpoints and to converge on an agreed upon course of action. This helps to keep issues focused and productive.
```
Comment (Josh-Cena) [inline files/en-us/mdn/community/issues/index.md:23]:
```suggestion
- Before filing an issue, consider if you need to start a [discussion](https://github.com/mdn/mdn-community/discussions) in the MDN Web Docs project on GitHub. Use discussions to gain different viewpoints and to converge on an agreed-upon course of action. This helps to keep issues focused and productive.
```

File: files/en-us/mdn/community/issues/index.md [hunk 3]
```diff
@@ -10,100 +10,203 @@ tags:
...
+
+#### Review the issue for completeness of information
+
+Review each issue against the following checklist to ensure that the issue contains the described information for someone to start working on the bug:
+
+- URL of the MDN Web Docs page with the problem or URL of an example MDN Web Docs page if the problem exists on multiple pages.
+- The specific heading or section on the MDN Web Docs page where the problem was found.
+- A clear description of the incorrect, unhelpful, incomplete, or missing information.
+
+If any of the above information is not present, then you should ask the author of the issue to provide these details and resume triaging the issue only after those details have been provided. It is okay to wait for up to a week to get a response from the author.
```
Comment (Josh-Cena) [inline files/en-us/mdn/community/issues/index.md:171]: Mention "needs info" label

File: files/en-us/mdn/community/issues/index.md [hunk 4]
```diff
@@ -10,100 +10,203 @@ tags:
...
+```
+
+#### Set other labels
+
+Next, set the following labels as appropriate:
+
+- `effort: small`, `effort: medium`, `effort: large`: Some contributors like to search for bugs based on the time and effort that will be needed to fix the bug. So where possible, you should try to provide an estimate of the required effort.
+- `good first issue`: Set this label on the issue if the fix for the issue is really simple and if fixing the issue would provide good practice for a newcomer who is getting used to the process.
+- `help wanted`: Set this label if the issue requires help from someone who knows about or is familiar with the topic. This is a popular label and some contributors use it to search for issues to work on in open source projects in their areas of familiarity or expertise.
+- `broken link internal`, `broken link external` — Set the appropriate label if the issue involves a link to a non-existent internal page or a broken external link.
```
Comment (Josh-Cena) [inline files/en-us/mdn/community/issues/index.md:209]: "non-existent internal page" generally uses the "document not written" label

**Comment by Josh-Cena** [comment]
@dipikabh I've merged the page for you. Are you on Linux?

**Comment by dipikabh** [comment]
> @dipikabh I've merged the page for you. I don't know what's wrong...

Wow, thanks a lot, Josh! I'm not sure why I was getting an error

File: files/en-us/mdn/community/issues/index.md [hunk 5]
```diff
@@ -10,100 +10,203 @@ tags:
...
+### Before filing an issue
 
-## Make progress, not noise
+If you think you've found a bug with the content on MDN Web Docs or with the look and feel of the website, search the current open issues in the [relevant repository](/en-US/docs/MDN/Community/Contributing/Our_repositories) and make sure nobody else has reported the issue.
 
-Think carefully about the way you handle communication in the project — make sure it is useful, and that it doesn't make other contributors jobs harder. Submitting pull requests to fix issues is great, but are they truly useful, and easy to review? Filing issues and joining in other conversations is fine, but are your issues and comments on topic, or are they just adding noise?
+### Reporting an issue
 
-As a rule, do:
+- Depending on the problem you've discovered, report it by filing a [documentation issue](https://github.com/mdn/content/issues), a [translation issue](https://github.com/mdn/translated-content/issues), or a [website look and feel issue](https://github.com/mdn/yari/issues).
```
[the remark this answers is not in the document]
Comment (dipikabh) [reply files/en-us/mdn/community/issues/index.md]: Makes sense. Done in https://github.com/mdn/content/pull/23479/commits/ebf65748ae260d655d17f2bc37d6cd658895b224.

File: files/en-us/mdn/community/issues/index.md [hunk 6]
```diff
@@ -10,100 +10,203 @@ tags:
...
 
-   - If someone else is already assigned to the issue:
+   > **Note:** An issue with the `needs triage` label indicates that the MDN Web Docs core team has not reviewed the issue yet, and you shouldn't begin work on it.
 
-     1. If this was more than one week ago, and there has not been much activity, @mention them and ask them if you can take it over, or otherwise help get it to completion.
+2. **Assign the issue to yourself:** After finding an issue you'd like to work on, make sure that the issue is not assigned to anybody else. Add a comment saying you would like to work on the issue, and if you are able to, [assign the issue to yourself](https://docs.github.com/en/issues/tracking-your-work-with-issues/assigning-issues-and-pull-requests-to-other-github-users#assigning-an-individual-issue-or-pull-request).
```
[the remark this answers is not in the document]
Comment (dipikabh) [reply files/en-us/mdn/community/issues/index.md:99]: I believe that is what this text is saying - only recommending to add a comment on the issue in addition to assigning the issue to oneself. This helps in "claiming" the issue part and letting other interested folks know the issue is being worked on by someone.

**Comment by dipikabh** [comment]
> @dipikabh I've merged the page for you. I don't know what's wrong...

@Josh-Cena, I think I figured what I was doing wrong. Removing the file as well as using the `yarn content delete` command, which is why the command could not find the file to be deleted.

**Review by bsmth — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**